src/schema.json

```diff
@@ -0,0 +1,96 @@
+{
+	"definitions": {
+		"mode": {
+			"type": "string",
+			"enum": [
+				"dev",
+				"dist",
+				"test"
+			]
+		},
+		"watch": {
+			"type": "string",
+			"enum": [
+				"memory",
+				"file"
+			]
+		},
+		"serve": {
+			"type": "boolean"
+		},
+		"port": {
+			"type": "number"
+		},
+		"legacy": {
+			"type": "boolean"
+		}
+	},
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"mode": {
+			"$ref": "#/definitions/mode"
+		},
+		"m": {
+			"$ref": "#/definitions/mode"
+		},
+		"watch": {
+			"$ref": "#/definitions/watch"
+		},
+		"w": {
+			"$ref": "#/definitions/watch"
+		},
+		"serve" : {
+			"$ref": "#/definitions/serve"
+		},
+		"s": {
+			"$ref": "#/definitions/serve"
+		},
+		"port" : {
+			"$ref": "#/definitions/port"
+		},
+		"p": {
+			"$ref": "#/definitions/port"
+		},
+		"legacy": {
+			"$ref": "#/definitions/legacy"
+		},
+		"l": {
+			"$ref": "#/definitions/legacy"
+		},
+		"prefix": {
+			"type": "string"
+		},
+		"widgets": {
```

Comment:
So this now support the mix of strings and objects in the array?

Reply:
Yeah this will support either a string or object with the file and tag properties (where file is required). You can mix and match in the array.